### PR TITLE
Add basic countdown component

### PR DESCRIPTION
**Pull request by l3satwik**

### CODE REVIEW

File: src/components/Countdown.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    timeRemaining: this.props.time
+  };
+
+  timer: NodeJS.Timeout | number | null = null;
+
+  componentDidMount() {
+    this.startCountdown();
+  }
+
+  startCountdown() {
```
Comment (azizhk) [inline src/components/Countdown.tsx]: private

File: src/components/Countdown.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+export default class Countdown extends React.PureComponent<
+  CountdownProps,
+  CountdownState
+> {
+  static defaultProps = {
+    time: 30,
+    counter: 1000
+  };
+
+  state = {
```
Comment (azizhk) [inline src/components/Countdown.tsx]:
```suggestion
  state: Readonly<CountdownState> = {
```

File: src/components/Countdown.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
+import * as React from "react";
+
+interface CountdownProps {
+  time: number; // in seconds
```
Comment (azizhk) [inline src/components/Countdown.tsx]:
```suggestion
  time?: number; // in seconds
```

File: src/components/Countdown.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  counter: number; // in seconds
```
Comment (azizhk) [inline src/components/Countdown.tsx]:
```suggestion
  counter?: number; // in seconds
```

File: src/components/Countdown.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  timeRemaining: number;
+}
+
+export default class Countdown extends React.PureComponent<
+  CountdownProps,
+  CountdownState
+> {
+  static defaultProps = {
+    time: 30,
+    counter: 1000
```
Comment (azizhk) [inline src/components/Countdown.tsx]: This does not look to be in seconds

File: src/components/Countdown.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  counter: number; // in seconds
+  onFinish: () => void;
+}
+
+interface CountdownState {
+  timeRemaining: number;
+}
+
+export default class Countdown extends React.PureComponent<
+  CountdownProps,
```
Comment (azizhk) [inline src/components/Countdown.tsx]:
```suggestion
  CountdownProps & typeof Countdown.defaultProps,
```

File: src/components/Countdown.tsx (hunk 3 above)
Comment (l3satwik) [reply src/components/Countdown.tsx]: Didn't we decide earlier that defaultProps should not be optional?

File: src/components/Countdown.tsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    this.clearInterval();
+  }
+
+  render() {
+    const { timeRemaining } = this.state;
+
+    const minutes = Math.floor(timeRemaining / 60);
+    const seconds = timeRemaining % 60;
+
+    return <>{`${minutes}:${seconds < 10 ? `0${seconds}` : seconds}`}</>;
```
Comment (ritz078) [inline src/components/Countdown.tsx]: you don't need to wrap it. You can return a string

**Comment by l3satwik** [comment]
Made the changes across web, closing this PR as it doesn't belong here, as discussed.

**Pull request closed without merging**